### ISSUE TITLE
LogFactory class not found

### ISSUE THREAD

**Issue by dseravalli** [Nordstrom/kafka-connect-lambda#34]
I get this error when I try to create a connector:

```
"java.lang.NoClassDefFoundError: org/apache/commons/logging/LogFactory\n\tat com.amazonaws.regions.AwsRegionProviderChain.<clinit>(AwsRegionProviderChain.java:33)\n\tat com.amazonaws.client.builder.AwsClientBuilder.<clinit>(AwsClientBuilder.java:60)\n\tat com.nordstrom.kafka.connect.lambda.InvocationClient$Builder.<init>(InvocationClient.java:126)\n\tat com.nordstrom.kafka.connect.lambda.InvocationClientConfig.<init>(InvocationClientConfig.java:55)\n\tat com.nordstrom.kafka.connect.lambda.LambdaSinkConnectorConfig.<init>(LambdaSinkConnectorConfig.java:52)\n\tat com.nordstrom.kafka.connect.lambda.LambdaSinkConnector.start(LambdaSinkConnector.java:31)\n\tat org.apache.kafka.connect.runtime.WorkerConnector.doStart(WorkerConnector.java:110)\n\tat org.apache.kafka.connect.runtime.WorkerConnector.start(WorkerConnector.java:135)\n\tat org.apache.kafka.connect.runtime.WorkerConnector.transitionTo(WorkerConnector.java:195)\n\tat org.apache.kafka.connect.runtime.Worker.startConnector(Worker.java:257)\n\tat org.apache.kafka.connect.runtime.distributed.DistributedHerder.startConnector(DistributedHerder.java:1183)\n\tat org.apache.kafka.connect.runtime.distributed.DistributedHerder.access$1400(DistributedHerder.java:125)\n\tat org.apache.kafka.connect.runtime.distributed.DistributedHerder$15.call(DistributedHerder.java:1199)\n\tat org.apache.kafka.connect.runtime.distributed.DistributedHerder$15.call(DistributedHerder.java:1195)\n\tat java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)\n\tat java.base/java.lang.Thread.run(Thread.java:834)\nCaused by: java.lang.ClassNotFoundException: org.apache.commons.logging.LogFactory\n\tat java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:471)\n\tat java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)\n\tat org.apache.kafka.connect.runtime.isolation.PluginClassLoader.loadClass(PluginClassLoader.java:104)\n\tat java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)\n\t... 18 more\n"
```

Connector config:
```
{
    "tasks.max": "1",
    "connector.class": "com.nordstrom.kafka.connect.lambda.LambdaSinkConnector",
    "topics": "<my-topic>",
    "key.converter": "org.apache.kafka.connect.storage.StringConverter",
    "value.converter": "org.apache.kafka.connect.storage.StringConverter",
    "aws.region": "us-east-1",
    "aws.lambda.function.arn": "arn:aws:lambda:us-east-1:<my-arn>",
    "aws.lambda.batch.enabled": "true",
    "aws.lambda.invocation.mode": "ASYNC",
    "aws.lambda.invocation.failure.mode": "DROP"
}
```

v1.2.2 of connector and Kafka Connect from Kafka v2.12-2.4.1 on Ubuntu 18.04.

**Comment by dylanmei** [comment]
I do not recognize this. Do you have other Connect plugins installed?

**Comment by dseravalli** [comment]
Yes I have the Confluent S3 and Lambda sinks both installed and working.

**Comment by SgtPepperLHCB** [comment]
`LogFactory` is a class found in the `commons-logging-1.2.jar` which is usually part of the Kakfa Connect distribution. For example, it can be found in `$CONFLUENT_HOME/share/java/kafka. common-logging-1.2.ja`r It is not included in the kafka-connect-lambda jar to avoid version conflict at runtime.

**Comment by dseravalli** [comment]
Makes sense. Is there anything I can troubleshoot to figure out why it wouldn't be finding the library? I am using some confluent connector plugins but I am running the community Kafka Connect binary. The confluent connector plugins are not distributed as uber jars and I notice that `common-loggings` is included in their `lib/` directories. I don't see it in my `kafka/libs` though.

**Comment by SgtPepperLHCB** [comment]
You are right. I don't see `commons-logging-1.2.jar` in the Apache tarball, but do see it in several places (including in connector directories) in a Confluent install.

As a work-around, you can put the logging jar file somewhere on your kafka-connect classpath, or in the same directory as the `kafka-connect-lambda.jar`.

We'll look into including the logging jar into a future release.

**Comment by dseravalli** [comment]
Thanks, that work-around worked

**Comment by dylanmei** [comment]
This is addressed with 1.3.0 release.
